packages/extension-ui/src/Popup/Accounts/EditAccountMenu.tsx

```diff
@@ -77,13 +73,18 @@ function EditAccountMenu({
     }
   }, [address, isHidden, account]);
 
-  const onAction = useContext(ActionContext);
-
   const goTo = useCallback((path: string) => () => onAction(path), [onAction]);
 
+  useEffect(() => {
+    if (account) {
+      setAccount(foundAccount);
+    }
+  }, [account, foundAccount]);
+
```

Comment:
I think the `isHidden` state might get desynchronized with `account` state if not set in here after `account` changes. Also, it's probably duplicated - `account.isHidden` should always be the same as the `isHidden` state, no?

---

On that topic - in case of this logic, I believe that a more semantic usage of hooks and state, leading to simpler data flow, would be to swap the states and derived data:
1. `isHidden` does not seem to be consumed as a part of the `account` object as a whole; the `account` object is merely a container for it, so `isHidden` can be easily extracted to a separate state - this enables us to visibly simplify things:
2. `account` does not have to be state (it being a state is problematic, because in that form it has to be updatable by both: prop change and internal state change, which obscures the flow)

I'm thinking something like this:
![image](https://user-images.githubusercontent.com/6209244/217353722-aae858be-461a-4e26-bc69-7c6a9b714832.png)


Last thing that this page might benefit from is to make the page remount after the `address` change (the page should react as if `key={address}` was set on it). Also it's more semantic - after all, a different address is a completely different account, so no reason to keep any state after the `address` change - cleaner to wipe it all.